### ISSUE TITLE
[BUG] TES will allow incorrect configuration

### ISSUE THREAD

**Issue by mitchwadair** [mitchwadair/tesjs#45]
**Describe the bug**
TES will still work if provided incorrect config (see #40).  This should not happen, and the constructor should throw errors for missing required properties.

**To Reproduce**
1. Supply an empty config object when initing TES
2. Attempt to subscribe to an event
3. See that TES has been inited and get an error when attempting to make the subscription request

**Expected behavior**
TES should throw an error on construction so the developer knows to use a valid config.

**Additional context**
Basically, need to prevent issues like #40

**Comment by mitchwadair** [comment]
I've already done some work on this on the `websockets-dev` branch, but it should probably make it into TES before that does.

**Comment by mitchwadair** [comment]
Closed by #48
